x-pack/plugins/runtime_fields/public/types.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { DataPublicPluginStart } from 'src/plugins/data/public';
+
+import { RUNTIME_FIELD_TYPES } from './constants';
+import { OpenRuntimeFieldEditorProps } from './load_editor';
+
+export interface LoadEditorResponse {
+  openEditor(props: OpenRuntimeFieldEditorProps): () => void;
+}
+
+export interface PluginSetup {
+  loadEditor(): Promise<LoadEditorResponse>;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface PluginStart {}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface SetupPlugins {}
+
+export interface StartPlugins {
+  data: DataPublicPluginStart;
+}
+
+export type RuntimeType = typeof RUNTIME_FIELD_TYPES[number];
+
+export interface RuntimeField {
+  name: string;
+  type: RuntimeType;
+  script: string;
+}
```

Comment:
Using this now as a type for what the editor returns, means we've decided against a higher level "intention" abstraction, that we would e.g. save later on "the user wanted to extract the weekday" instead of the full script? Could you please elaborate a bit on why we made the decicions to always store a script?

Reply:
At this stage, we haven't decided against anything yet 😊 What we do know for sure is that we will need to be able to return the raw script whenever the user wants to go that way and he will bypass any UI we create to help him build the script. And this is the first iteration of the editor, it only allows to write a script.
When we will improve on the UX to create the script we can decide how best to store the user intention 👍

Reply:
@timroes @sebelga Maybe we can split the difference. We want something that will be forward compatible but we don't necessarily want to fully define it or block progress on this pr. 

I'll toss out an idea - what if a future version of `RuntimeField` had an `editor` attribute as so - 

```
export interface RuntimeFieldEditor {
  id: string; // or something more specific
  params: { [key: string]: any; }
}
```

lack of an editor param would indicate the generic editor we're currently working on.

Reply:
This could work, but I was more seeing something like a "script definition" object. If present, then the script is generated.

```js
interface RuntimeField {
  name: string;
  type: string;
  script?: string; // optional but either this or scriptDefinition must be present
  scriptDefinition?: {
    // To be defined! :)
    dateParse: {
      target: '_doc',
      field: 'eventDate',
      ... 
    },
    stringConcat: {
      ...
    }
  }
}
```

But it's a bit early to define the interface IMO.